lib/Models/UserDrawing.js

```diff
@@ -0,0 +1,418 @@
+'use strict';
+
+var MapInteractionMode = require('../Models/MapInteractionMode');
+
+var DeveloperError = require('terriajs-cesium/Source/Core/DeveloperError');
+var defined = require('terriajs-cesium/Source/Core/defined');
+var defaultValue = require('terriajs-cesium/Source/Core/defaultValue');
+var Color = require('terriajs-cesium/Source/Core/Color');
+var PolylineGlowMaterialProperty = require('terriajs-cesium/Source/DataSources/PolylineGlowMaterialProperty');
+var CustomDataSource = require('terriajs-cesium/Source/DataSources/CustomDataSource');
+var CallbackProperty = require('terriajs-cesium/Source/DataSources/CallbackProperty');
+var PolygonHierarchy = require('terriajs-cesium/Source/Core/PolygonHierarchy');
+var knockout = require('terriajs-cesium/Source/ThirdParty/knockout');
+var Entity = require('terriajs-cesium/Source/DataSources/Entity.js');
+var when = require('terriajs-cesium/Source/ThirdParty/when');
+var Cartesian3 = require('terriajs-cesium/Source/Core/Cartesian3');
+
+/**
+ * Callback for when a point is clicked.
+ * @callback onPointClicked
+ * @param {CustomDataSource} customDataSource Contains all point entities that user has selected so far
+ */
+
+/**
+ * Callback for when clean up is happening, i.e., for done or cancel.
+ * @callback onCleanUp
+ */
+
+/**
+ * Callback for when the dialog is displayed, to provide a custom message
+ * @callback onMakeDialogMessage
+ * @return {String} Message to add to dialog
+ */
+
+/**
+ * @alias UserDrawing
+ * @constructor
+ *
+ * For user drawings, which includes lines and/or a polygon
+ * @param {Object} options Object with the following properties, all of which are optional except terria:
+ * @param {Terria} [options.terria] The Terria instance.
```

Comment:
The brackets around `options.terria` mean the parameter is optional.  Remove them for required parameters, and you can also remove the comment above.  You can specify what default value is used when an optional parameter isn't supplied with something like `[options.allowPolygon=true]`.